src/coprocessor/dag/expr/arithmetic.rs

```diff
@@ -197,11 +197,94 @@ impl FnCall {
             None => Ok(None),
         }
     }
+
+    pub fn intdivide_int(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let lhs = try_opt!(self.children[0].eval_int(ctx, row));
```

Comment:
check the divisor first like `tidb` did, it has better performance when the divisor is 0.

Reply:
you're right, thanks!